### PR TITLE
Refactoring named tuple

### PR DESCRIPTION
**Pull request by noelladsa**
Introduced three namedtuples along with associated refactoring
LinePart - Representing a word and its location in the interpreter input
FuncProps - Replaces the current argspec which contained function specific info along with Arg info. 
Argspec - Newly defined Argspec representing python2 and python3 argspec result from the inspect.getargspec()

### CODE REVIEW

File: bpython/autocomplete.py [hunk 1]
```diff
@@ -2,7 +2,6 @@
 
 # The MIT License
 #
-# Copyright (c) 2009-2015 the bpython authors.
```
Comment (sebastinas) [inline bpython/autocomplete.py]: No, please keep that line in.

**Comment by sebastinas** [comment]
Thanks for cleaning up the code base. Unfortunately the test suite fails with the proposed changes. Please take a look at it.

**Comment by noelladsa** [comment]
Yes, just realized that nosetests was reporting no errors but I hadn't installed twisted so those tests were skipped. Shall fix and do another pull. Thanks for the review.

File: bpython/inspection.py [hunk 2]
```diff
@@ -40,6 +41,11 @@
 
     _name = LazyReCompile(r'[a-zA-Z_]\w*$')
 
+ArgSpec = namedtuple('ArgSpec', ['args', 'varargs', 'varkwargs',  'defaults',
+                        'kwonly', 'kwonly_defaults', 'annotations'])
+
```
Comment (sebastinas) [inline bpython/inspection.py]: Please align this line correctly (`'kwonly'` should be exactly below `'args'`).

File: bpython/autocomplete.py [hunk 3]
```diff
@@ -126,8 +127,8 @@ def format(self, word):
 
     def substitute(self, cursor_offset, line, match):
         """Returns a cursor offset and line with match swapped in"""
-        start, end, word = self.locate(cursor_offset, line)
-        result = start + len(match), line[:start] + match + line[end:]
+        lpart =  self.locate(cursor_offset, line)
+        result = lpart.start + len(match), line[:lpart.start] + match + line[lpart.end:]
```
Comment (sebastinas) [inline bpython/autocomplete.py]: Please break lines at 80 characters.

File: bpython/autocomplete.py [hunk 4]
```diff
@@ -405,7 +401,7 @@ def matches(self, cursor_offset, line, **kwargs):
                 # if identifier isn't ascii, don't complete (syntax error)
                 if word is None:
                     continue
-                if self.method_match(word, n, text) and word != "__builtins__":
+                if self.method_match(word, n, r.word) and word != "__builtins__":
```
Comment (sebastinas) [inline bpython/autocomplete.py]: Please break lines at 80 characters.

File: bpython/curtsiesfrontend/replpainter.py [hunk 5]
```diff
@@ -159,13 +157,13 @@ def formatted_docstring(docstring, columns, config):
...
-def paint_infobox(rows, columns, matches, argspec, match, docstring, config,
+def paint_infobox(rows, columns, matches, funcprops, arg_pos, match, docstring, config,
                   format):
-    """Returns painted completions, argspec, match, docstring etc."""
+    """Returns painted completions, funcprops, match, docstring etc."""
     if not (rows and columns):
         return fsarray(0, 0)
     width = columns - 4
-    lines = ((formatted_argspec(argspec, width, config) if argspec else []) +
+    lines = ((formatted_argspec(funcprops, arg_pos, width, config) if funcprops else []) +
```
Comment (sebastinas) [inline bpython/curtsiesfrontend/replpainter.py]: Please break the line at 80 characters.

File: bpython/repl.py [hunk 6]
```diff
@@ -468,8 +469,8 @@ def get_object(self, name):
 
     def get_args(self):
         """Check if an unclosed parenthesis exists, then attempt to get the
-        argspec() for it. On success, update self.argspec and return True,
-        otherwise set self.argspec to None and return False"""
+        argspec() for it. On success, update self.funcprops,self.arg_pos and return True,
```
Comment (sebastinas) [inline bpython/repl.py]: Please break the line at 80 characters.

**Comment by sebastinas** [comment]
Looks good except a few issues found by pep8. Could you fix them too?

**Comment by noelladsa** [comment]
yes, will do.

Thanks!

On Tue, Jun 9, 2015 at 4:40 PM, Sebastian Ramacher <notifications@github.com

**Pull request merged**